File: src (cpp)/animal/mammalia/lion.cpp
#include "lion.h"

Lion::Lion(int _x, int _y, int _weight) : def_ID(2),
                                          def_ratio_meat(60),
                                          def_ratio_plant(0) {
  ID = def_ID;
  position.SetX(_x);
  position.SetY(_y);
  ratio_meat = def_ratio_meat;
  ratio_plant = def_ratio_plant;
  is_land_animal = true;
  is_water_animal = false;
  is_air_animal = false;
  weight = _weight;
}
void Lion::Interact() {
  std::cout << "The lion is roaring" << std::endl;
}

File: src/virtualzoo/animal/mammalia/Lion.java
package virtualzoo.animal.mammalia;

import virtualzoo.animal.behavior.BehaviorTame;
import virtualzoo.animal.diet.Carnivora;

/**
 * Kelas Lion mendefinisikan atribut untuk lion.
 *
 * @author Rionaldi Chandraseta - 13515077
 * @version 3.0
 * @since 3.0
 */
public class Lion extends Mammalia {

  /**
   * Constructor
   *
   * Menciptakan lion dengan posisi (locX,locY) dan berat animalWeight.
   *
   * @param locX Nilai absis posisi.
   * @param locY Nilai ordinat posisi.
   * @param animalWeight Nilai berat lion.
   */
  public Lion(int locX, int locY, int animalWeight) {
    super();
    animalBehavior = new BehaviorTame();
    animalDiet = new Carnivora(animalWeight, 60);
    ID = 2;
    position.setLocation(locX, locY);
    isLandAnimal = true;
    isWaterAnimal = false;
    isAirAnimal = false;
  }

  /**
   * Mengatur interaksi dengan lion.
   */
  public void interact() {
    System.out.println("The lion is roaring");
  }
}